File: Go/TCPServer.go
//Includes
 package main
 import "fmt"
 import "net"
func main(){
	server,err:=net.Listen("tcp",":3721")
	if(err!=nil){panic("Failed listen")}
	for{
	 client,err:=server.Accept()
	 if(err!=nil){panic("Failed accept")}
	 buf:=make([]byte,256)
	 size,err:=client.Read(buf)
	 if(err!=nil){panic("Failed TCP read")}
	 fmt.Print(string(buf[:size]))
	 client.Write([]byte("Message received\n"));
	 client.Close();}}

File: TCPServer/Javascript/TCPServer.js
const net=require("net")
//I couldn't figure out how to set address reuse so it isn't done here
const server=net.createServer(client=>
{
 client.on('data',data=>
 {
  process.stdout.write(data.toString('ascii')) //When a message is recieved write it to console
 })
 client.write("Message received\n")            //Send reply
 client.close()                                //Close the client
})
server.listen(3721) //Start listening on port 3721